File: app/src/main/java/com/theone/music/data/constant/NetConstant.java
package com.theone.music.data.constant;//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import java.util.HashMap;
import java.util.Map;

import rxhttp.wrapper.annotation.DefaultDomain;

/**
 * @author The one
 * @date 2022-04-11 11:05
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
public class NetConstant {

    @DefaultDomain
    public static final String BASE_URL = "https://www.hifini.com/";

    public static final String INDEX = "index-%d.htm";
    public static final String FORUM = "forum-%d-%d.htm";
    public static final String SEARCH = "search-%s.htm";
    public static final String HOT = "index-0-2.htm";
    public static final String RANK = "index-0-%d.htm";
    public static final String TAG = "tag-%d-%d.htm";

    private static Map<String,Integer> CATEGORY;

    public static Map<String,Integer> getCategoryList(){
        if(null == CATEGORY){
            CATEGORY = new HashMap<>();
            CATEGORY.put("华语",1);
            CATEGORY.put("日韩",15);
            CATEGORY.put("欧美",10);
            CATEGORY.put("Remix",11);
            CATEGORY.put("纯音乐",12);
            CATEGORY.put("异次元",13);
        }
        return CATEGORY;
    }

    private static Map<String,Integer> RANK_TYPES;

    public static Map<String,Integer> getRankTypes(){
        if(null == RANK_TYPES){
            RANK_TYPES = new HashMap<>();
            RANK_TYPES.put("日榜",5);
            RANK_TYPES.put("周榜",4);
            RANK_TYPES.put("月榜",3);
        }
        return RANK_TYPES;
    }

}

File: app/src/main/java/com/theone/music/net/NetConstant.kt
package com.theone.music.net

import rxhttp.wrapper.annotation.DefaultDomain

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2022-01-04 11:30
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
object NetConstant {

    @DefaultDomain
    const val BASE_URL = "https://www.hifini.com/"

    /**
     * 登录 - 拿取[cookie]里的[bbs_token]
     */
    const val LOGIN = "user-login.htm"

    /**
     * 注册
     * @POST
     * @param email
     * @param username
     * @param password
     * @param repeat_password
     */
    const val REGISTER = "user-create.htm"

    /**
     * 修改头像
     * @POST
     * @param width
     * @param height
     * @param action
     * @param filetype  jpg
     * @param name xxx.png
     * @param data file
     */
    const val AVATAR = "my-avatar.htm"

    /**
     * 签到
     */
    const val SIGN = "sg_sign.htm"

    /**
     * 用户信息
     */
    const val USER_INFO = "my.htm"


    const val INDEX = "index-%d.htm"

    const val FORUM = "forum-%d-%d.htm"

    const val SEARCH = "search-%s.htm"

    const val HOT = "index-0-2.htm"

    const val RANK = "index-0-%d.htm"

    const val TAG = "tag-%d-%d.htm"

    val CATEGORY = mapOf<String,Int>("华语" to 1,"日韩" to 15,"欧美" to 10,"Remix" to 11,"纯音乐" to 12,"异次元" to 13)

    val RANK_TYPES = mapOf<String,Int>("日榜" to 5,"周榜" to 4,"月榜" to 3)

}